larq_zoo/xnornet.py

```diff
@@ -0,0 +1,203 @@
+from zookeeper import registry, HParams
+import larq as lq
+import tensorflow as tf
+from larq_zoo import utils
+
+
+@registry.register_model
+def xnornet(hparams, dataset, input_tensor=None, include_top=True):
+    kwargs = dict(
+        kernel_quantizer=hparams.kernel_quantizer,
+        input_quantizer=hparams.input_quantizer,
+        kernel_constraint=hparams.kernel_constraint,
+        use_bias=hparams.use_bias,
+        kernel_regularizer=hparams.kernel_regularizer,
+    )
+    img_input = utils.get_input_layer(dataset.input_shape, input_tensor)
+
+    x = tf.keras.layers.Conv2D(
+        96,
+        (11, 11),
+        strides=(4, 4),
+        padding="same",
+        use_bias=hparams.use_bias,
+        input_shape=dataset.input_shape,
+        kernel_regularizer=hparams.kernel_regularizer,
+    )(img_input)
+
+    x = tf.keras.layers.BatchNormalization(
+        momentum=hparams.bn_momentum, scale=hparams.bn_scale, epsilon=1e-5
+    )(x)
+    x = tf.keras.layers.Activation("relu")(x)
+    x = tf.keras.layers.MaxPool2D(pool_size=(3, 3), strides=(2, 2))(x)
+    x = tf.keras.layers.BatchNormalization(
+        momentum=hparams.bn_momentum, scale=hparams.bn_scale, epsilon=1e-4
+    )(x)
+    x = lq.layers.QuantConv2D(256, (5, 5), strides=(1, 1), padding="same", **kwargs)(x)
+    x = tf.keras.layers.MaxPool2D(pool_size=(3, 3), strides=(2, 2))(x)
+    x = tf.keras.layers.BatchNormalization(
+        momentum=hparams.bn_momentum, scale=hparams.bn_scale, epsilon=1e-4
+    )(x)
+    x = lq.layers.QuantConv2D(384, (3, 3), strides=(1, 1), padding="same", **kwargs)(x)
+    x = tf.keras.layers.BatchNormalization(
+        momentum=hparams.bn_momentum, scale=hparams.bn_scale, epsilon=1e-4
+    )(x)
+    x = lq.layers.QuantConv2D(384, (3, 3), strides=(1, 1), padding="same", **kwargs)(x)
+    x = tf.keras.layers.BatchNormalization(
+        momentum=hparams.bn_momentum, scale=hparams.bn_scale, epsilon=1e-4
+    )(x)
+    x = lq.layers.QuantConv2D(256, (3, 3), strides=(1, 1), padding="same", **kwargs)(x)
```

Comment:
I would rather be as explicit as possible for readibility

Reply:
I would also be in favour of removing it, readability doesn't improve with unnecessary details imo

Reply:
> I would also be in favour of removing it, readability doesn't improve with unnecessary details imo

I agree. In general I also like to be explicit, but in this case it makes it harder for me to quickly grasp whats happening in this layer since I have to think if strinding is something special here or if it is just the default that's repeated.

Reply:
SLC